UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -0,0 +1,654 @@
+          Data32 = (UINT32 *)(PropertyPtr->Data);
+          // DEBUG ((DEBUG_INFO, "\n         Property(%08X)  %a", Property, TempStr));
+          // DEBUG ((DEBUG_INFO, "  %X", Fdt32ToCpu(*Data32)));
+          GraphicsInfo->GraphicsMode.PixelInformation.RedMask = Fdt32ToCpu (*Data32);
+          // DEBUG ((DEBUG_INFO, "  %X", Fdt32ToCpu(*(Data32 + 1))));
+          GraphicsInfo->GraphicsMode.PixelInformation.GreenMask = Fdt32ToCpu (*(Data32 + 1));
+          // DEBUG ((DEBUG_INFO, "  %X", Fdt32ToCpu(*(Data32 + 2))));
+          GraphicsInfo->GraphicsMode.PixelInformation.BlueMask = Fdt32ToCpu (*(Data32 + 2));
+        } else if (AsciiStrCmp (TempStr, "pixe-scanline") == 0) {
+          Data32 = (UINT32 *)(PropertyPtr->Data);
+          // DEBUG ((DEBUG_INFO, "\n         Property(%08X)  %a", Property, TempStr));
+          // DEBUG ((DEBUG_INFO, "  %X", Fdt32ToCpu(*Data32)));
+          GraphicsInfo->GraphicsMode.PixelsPerScanLine = Fdt32ToCpu (*Data32);
+        } else if (AsciiStrCmp (TempStr, "id") == 0) {
+          Data32 = (UINT32 *)(PropertyPtr->Data);
+          // DEBUG ((DEBUG_INFO, "\n         Property(%08X)  %a", Property, TempStr));
+          // DEBUG ((DEBUG_INFO, "  %X", Fdt32ToCpu(*Data32)));
+          GraphicsDev->VendorId = Fdt32ToCpu (*Data32) >> 16;
+          GraphicsDev->DeviceId = Fdt32ToCpu (*Data32) & 0xffff;
+        } else if (AsciiStrCmp (TempStr, "subsystem-id") == 0) {
+          Data32 = (UINT32 *)(PropertyPtr->Data);
+          // DEBUG ((DEBUG_INFO, "\n         Property(%08X)  %a", Property, TempStr));
+          // DEBUG ((DEBUG_INFO, "  %X", Fdt32ToCpu(*Data32)));
+          GraphicsDev->SubsystemVendorId = Fdt32ToCpu (*Data32) >> 16;
+          GraphicsDev->SubsystemId       = Fdt32ToCpu (*Data32) & 0xffff;
+        } else if (AsciiStrCmp (TempStr, "revision-id") == 0) {
+          Data32 = (UINT32 *)(PropertyPtr->Data);
+          // DEBUG ((DEBUG_INFO, "\n         Property(%08X)  %a", Property, TempStr));
+          // DEBUG ((DEBUG_INFO, "  %X", Fdt32ToCpu(*Data32)));
+          GraphicsDev->RevisionId = (UINT8)Fdt32ToCpu (*Data32);
+        } else if (AsciiStrCmp (TempStr, "bar-index") == 0) {
+          Data32 = (UINT32 *)(PropertyPtr->Data);
+          // DEBUG ((DEBUG_INFO, "\n         Property(%08X)  %a", Property, TempStr));
+          // DEBUG ((DEBUG_INFO, "  %X", Fdt32ToCpu(*Data32)));
+          GraphicsDev->BarIndex = (UINT8)Fdt32ToCpu (*Data32);
+        }
+      }
+    } else if (AsciiStrCmp (NodePtr->Name, "cpu-info") == 0) {
+      PropertyPtr = FdtGetProperty (Fdt, Node, "address_width", &TempLen);
+      ASSERT (TempLen > 0);
+      if (TempLen > 0) {
+        Data32 = (UINT32 *)(PropertyPtr->Data);
+        DEBUG ((DEBUG_INFO, "\n         Property(00000000)  address_width"));
+        DEBUG ((DEBUG_INFO, "  %X", Fdt32ToCpu (*Data32)));
+        SizeOfMemorySpace = (UINT8)Fdt32ToCpu (*Data32);
+
+        BuildCpuHob (SizeOfMemorySpace, 16);

```

Comment:
16 is an assumption made for all archs?

Reply:
Basing on the discussion today, we would like to use #if defined (MDE_CPU_IA32) || defined (MDE_CPU_X64) for 16 bit iospace support. Otherwise this will be 0 so no iospace support.